src/utils/pgwire/src/pg_message.rs

```diff
@@ -332,8 +332,14 @@ impl FeStartupMessage {
     pub async fn read(stream: &mut (impl AsyncRead + Unpin)) -> Result<FeMessage> {
         let len = stream.read_i32().await?;
         let protocol_num = stream.read_i32().await?;
-        let payload_len = len - 8;
-        let mut payload = vec![0; payload_len as usize];
+        let payload_len = (len - 8) as usize;
+        if payload_len >= isize::MAX as usize {
```

Comment:
I think it's better to make this a configuration rather a constant

Reply:
This comes from here: https://doc.rust-lang.org/src/alloc/raw_vec.rs.html#505

Actually i don't think we should make it a config, cuz even devs are not expected to roll it down or up in most cases, not talk about users. This is just avoid the capacity overflow problem.